src/status_im/contexts/communities/overview/view.cljs

```diff
@@ -137,119 +109,112 @@
-           #(join-gated-community id))
-         :accessibility-label :join-community-button
-         :customization-color color
-         :container-style {:margin-horizontal 12 :margin-top 12 :margin-bottom 12}
-         :disabled? (not can-request-access?)
-         :icon-left (if can-request-access? :i/unlocked :i/locked)}
-        (i18n/label :t/join-open-community)]])))
+(defn- token-requirements
+  [{:keys [id color]}]
+  (let [{:keys [can-request-access?
+                number-of-hold-tokens tokens
+                highest-permission-role]} (rf/sub [:community/token-gated-overview id])
+        highest-role-text
+        (i18n/label
+         (communities.utils/role->translation-key highest-permission-role :t/member))]
+    [rn/view {:style (style/token-gated-container)}
+     [rn/view
+      {:style {:padding-horizontal 12
+               :flex-direction     :row
+               :align-items        :center
+               :justify-content    :space-between
+               :flex               1}}
+      [quo/text {:weight :medium}
+       (if can-request-access?
+         (i18n/label :t/you-eligible-to-join-as {:role highest-role-text})
+         (i18n/label :t/you-not-eligible-to-join))]
+      [info-button]]
+     (when (pos? number-of-hold-tokens)
+       [quo/text {:style {:padding-horizontal 12 :padding-bottom 18} :size :paragraph-2}
+        (if can-request-access?
+          (i18n/label :t/you-hold-number-of-hold-tokens-of-these
+                      {:number-of-hold-tokens number-of-hold-tokens})
+          (i18n/label :t/you-must-hold))])
+     [quo/token-requirement-list
+      {:tokens   tokens
+       :padding? true}]
+     [quo/button
+      {:on-press            #(rf/dispatch [:open-modal :community-account-selection
+                                           {:community-id id}])
+       :accessibility-label :join-community-button
+       :customization-color color
+       :container-style     {:margin-horizontal 12 :margin-top 12 :margin-bottom 12}
+       :disabled?           (not can-request-access?)
+       :icon-left           (if can-request-access? :i/unlocked :i/locked)}
+      (i18n/label :t/join-open-community)]]))
 
-(defn join-community
-  [{:keys [joined color permissions token-permissions membership-permissions? id] :as community}
-   pending?]
-  (let [access-type     (get-access-type (:access permissions))
+(defn- join-community
+  [{:keys [id color joined permissions role-permissions? can-join?] :as community}]
+  (let [pending?        (rf/sub [:communities/my-pending-request-to-join id])
+        access-type     (get-access-type (:access permissions))
         unknown-access? (= access-type :unknown-access)
         invite-only?    (= access-type :invite-only)]
     [:<>
      (when-not (or joined pending? invite-only? unknown-access?)
-       (if membership-permissions?
-         [token-gates community]
+       (if role-permissions?
```

Comment:
FYI: _Role permissions_ is the name we have found to better represent what in Figma is named _Token Gated_. We believe _Token Gated_ doesn't correctly capture the intent of the permissions because a community can have non-permissions that also "token gate" the community.